### PR TITLE
Pulling in fixes from other forks

### PR DESCRIPTION
**Pull request by chris18890**
Pulling in fixes from @mjgardes & @rumplestilzken

### CODE REVIEW

File: crm/modules/payment/payment.inc.php [hunk 1]
```diff
@@ -350,7 +350,7 @@ function payment_data ($opts = array()) {
                     $clause .= "`created` ";
                     break;
                 default:
-                    continue;
+                    continue 2;
```
Comment (joshprittse) [inline crm/modules/payment/payment.inc.php:353]: What does the 2 here do?

**Review by joshprittse — COMMENTED** [review]
Quick inline question

File: crm/modules/payment/payment.inc.php (hunk 1 above)
Comment (chris18890) [inline crm/modules/payment/payment.inc.php:353]: As I understand it, it forces the continuation of the loop - https://www.php.net/manual/en/control-structures.continue.php

Comment (rumplestilzken) [reply crm/modules/payment/payment.inc.php:353]: Yes, continue at this point in the code will act as a break, you need continue 2 to force the next iteration of the loop.

**Pull request merged**